client/app/components/Timer.jsx

```diff
@@ -0,0 +1,40 @@
+import moment from 'moment';
+import { useState, useMemo, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { react2angular } from 'react2angular';
+import { Moment } from '@/components/proptypes';
+
+function timeFrom(startTime) {
+  return moment.utc(moment.now() - startTime).format('HH:mm:ss');
+}
+
+export function Timer({ from }) {
+  const startTime = useMemo(() => moment(from).valueOf(), [from]);
```

Comment:
Using `useMemo` as a sort of `componentDidMount` alternative?

Reply:
@ranbena  no, it's rather `getDerivedStateFromProps` - I need to prepare start time using `from` prop, but don't want to do it each time when component is re-rendered.